src/Log/Engine/BaseLog.php

```diff
@@ -94,6 +113,18 @@ public function scopes()
      * @return string
      */
     protected function _format(string $message, array $context = []): string
+    {
+        return $this->resolve($message, $context);
+    }
+
+    /**
+     * Resolves interpolation expressions in message string.
+     *
+     * @param string $message Interpolated message
+     * @param array $context Interpolation expression values
+     * @return string
+     */
+    protected function resolve(string $message, array $context = []): string
```

Comment:
Split resolving interpolate strings from _format() so we can deprecated _format() in the future and provide engines with this common functionality.

Existing log engines require the message to be resolved before formatted.